sage-ios/SAGE/SAGE/AnnouncementsViewController.swift

```diff
@@ -14,6 +14,41 @@ class AnnouncementsViewController: UITableViewController {
     var activityIndicator: UIActivityIndicatorView = UIActivityIndicatorView(activityIndicatorStyle: .Gray)
     var currentErrorMessage: ErrorView?
     
+    
+    override init(style: UITableViewStyle) {
+        super.init(style: style)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "addAnnouncement:", name: NotificationConstants.addAnnouncementKey, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "editAnnouncement:", name: NotificationConstants.editAnnouncementKey, object: nil)
+    }
+    
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func addAnnouncement(notification: NSNotification) {
```

Comment:
can you rename to like announcementAdded instead? or else it sounds like an action done by this controller